lib/smart_app_launch/standalone_launch_group.rb

```diff
@@ -95,8 +95,32 @@ class StandaloneLaunchGroup < Inferno::TestGroup
       }
     )
 
+    test from: :tls_version_test,
+         id: :standalone_auth_tls,
+         title: 'OAuth 2.0 authorize endpoint secured by transport layer security',
+         description: %(
+           Apps MUST assure that sensitive information (authentication secrets,
+           authorization codes, tokens) is transmitted ONLY to authenticated
+           servers, over TLS-secured channels.
+         ),
+         config: {
+           inputs: { url: { name: :smart_authorization_url } },
+           options: {  minimum_allowed_version: OpenSSL::SSL::TLS1_2_VERSION }
+         }
     test from: :smart_app_redirect
     test from: :smart_code_received
+    test from: :tls_version_test,
+         id: :standalone_token_tls,
+         title: 'OAuth 2.0 token endpoint secured by transport layer security',
+         description: %(
+           Apps MUST assure that sensitive information (authentication secrets,
+           authorization codes, tokens) is transmitted ONLY to authenticated
+           servers, over TLS-secured channels.
+         ),
+         config: {
+           inputs: { url: { name: :smart_token_url } },
+           options: {  minimum_allowed_version: OpenSSL::SSL::TLS1_2_VERSION }
```

Comment:
Does this also reject tls<1.2? Hard to tell by name. Also, are we planning on just having smart always follow the more strict onc guidlelines to keep it simpler (question depends on if this is following those anyhow, which they may or may not be).

I would think if smart is more lenient the non-shortcut way of doing this without leaking requirements across boundaries is we have this test be more lenient and then add an additional tls test at the g10 level to check the additional constraint there. Just a matter of how much we want to respect independence and separation.

Reply:
I feel like it's a bit ambiguous. The only language in the 1.0 spec is:

> Apps SHALL ensure that sensitive information (authentication secrets, authorization codes, tokens) is transmitted ONLY to authenticated servers, over TLS-secured channels.

Does using a deprecated, insecure version of TLS count as being "TLS-secured"? As you say, though, it's easy to make this lenient and have a stricter requirement in the g10 kit.